storage/sealer/tarutil/systar.go

```diff
@@ -6,45 +6,118 @@ import (
 	"io/ioutil"
 	"os"
 	"path/filepath"
+	"strings"
 
 	logging "github.com/ipfs/go-log/v2"
 	"golang.org/x/xerrors"
 )
 
 var log = logging.Logger("tarutil") // nolint
 
-func ExtractTar(body io.Reader, dir string, buf []byte) error {
+var CacheFileConstraints = map[string]int64{
```

Comment:
Should such a map belong in code or some sort of config?

Reply:
Those files have well-known names and sizes because they are generated in the sealing process which is tied to how the proving system works, so there isn't much to configure here.

The main point of those constraints is to protect against "zip-slip", and to make it harder for rouge SaaS to waste all disk space with invalid caches.